plugins/browser-plugin-media-tracking/src/api.ts

```diff
@@ -0,0 +1,81 @@
+import { Logger } from '@snowplow/tracker-core';
+import { BrowserPlugin, BrowserTracker } from '@snowplow/browser-tracker-core';
+import { waitForElement } from './findElem';
+import { Config, isElementConfig, isStringConfig } from './config';
+import { setUpListeners } from './player';
+import { setConfigDefaults } from './helperFunctions';
+
+// These imports are used for documentation purposes only.
+// Typescript complains that they are unused.
+// @ts-ignore: TS6133
+import { DynamicContext } from '@snowplow/tracker-core';
+// @ts-ignore: TS6133
+import { HTML5MediaEventTypes } from './config';
+// @ts-ignore: TS6133
+import { FilterOutRepeatedEvents } from '@snowplow/browser-plugin-media/src/types';
+import { endMediaTracking as endMediaTrackingPlugin } from '@snowplow/browser-plugin-media';
+
+let LOG: Logger;
+const _trackers: Record<string, BrowserTracker> = {};
+
+export function MediaTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+    logger: (logger) => {
+      LOG = logger;
+    },
+  };
+}
+
+/**
+ * Enables media tracking on an HTML `<video>` or `<audio>` element.
+ *
+ * @param {Config} config - Configuration options for media tracking.
+ *
+ * **Required:**
+ * - `config.id` (`string`): The session ID. Must be unique for each media element. Used to identify the media session and end tracking.
+ * - `config.video` (string | HTMLMediaElement): The ID of the media element or the element itself.
+ *
+ * **Optional:**
+ * - `config.label` (`string`): A human-readable label for the media element.
+ * - `config.captureEvents` {@link HTML5MediaEventTypes}: A list of media events to track. All events are tracked by default.
+ * - `config.boundaries` (`number[]`): Percentage thresholds (0-100) at which to trigger progress events. Disabled by default.
+ * - `config.context` {@link DynamicContext}: Contexts to attach to each tracking event.
+ *
+ * **Ping Configuration (Optional):**
+ * - `config.pings.pingInterval` (`number`): The interval (in seconds) for sending ping events. Default is 30(s).
+ * - `config.pings.maxPausedPings` (`number`): The maximum number of ping events sent while playback is paused. Default is 1.
+ *
+ * **Other Options (Optional):**
+ * - `config.updatePageActivityWhilePlaying` (`boolean`): Whether to update page activity while media is playing. Enabled by default.
+ * - `config.filterOutRepeatedEvents` {@link FilterOutRepeatedEvents}: Whether to suppress consecutive identical events. Default is false.
+ */
+export function startMediaTracking(config: Config) {
```

Comment:
Sorry I may have misled you – I think now we have a clash between the `startMediaTracking` function in this plugin and the one in the media plugin. I think that is the reason why the integration tests don't work for me – when I look into the compiled `sp.js` function, I can see that it is the one from the other media plugin.

Can we rename this function to `startHtml5MediaTracking` (or something similar) and also rename the `endMediaTracking` one?

Reply:
You can also rebase on top of the `release/4.0.0` branch which disables the youtube integration tests that are failing here as welll.